### ISSUE TITLE
interceptors.response async support

### ISSUE THREAD

**Issue by rhobot** [alex-cory/use-http#214]
Hi! I have a question/request regarding `interceptors.response`.

I've noticed that making `interceptors.response` function as async doesn't work. But there are some use cases where awaiting calls in the function is useful. For example, I would like to pre-parse `json` in the interceptor.

```js
<Provider options={{
  interceptors: {
    async response(response) {
      // ...
      response.parsedBody = await response.json() 
      return response
    }
  }
}} ...>
```
I was wondering whether async is supported for `interceptors.response` or it will be supported in the future. Or I was wondering this is by design.

Thanks.

**Comment by alex-cory** [comment]
I like this idea and am going to implement it.

Btw, `parsedBody` is essentially the same thing as `response.data`

**Comment by alex-cory** [comment]
This is implemented in `v0.4.1`. Closing this issue. Let me know if the issue persists, and I'll re-open the issue.

**Comment by TannerS** [comment]
> I like this idea and am going to implement it unless you want to submit a PR!
> 
> Btw, `parsedBody` is essentially the same thing as `response.data`.
> 
> Basically `response.data = await response.json() || await response.text()`

Hey, not to bring up a dead issue but how do i specifically call response.text() on a response?
